src/System.Runtime.Extensions/src/System/IO/StreamWriter.cs

```diff
@@ -446,6 +447,21 @@ public override void Write(char[] buffer, int index, int count)
             }
         }
 
+        public override void Write(ReadOnlySpan<char> source)
+        {
+            char[] buffer = ArrayPool<char>.Shared.Rent(source.Length);
+
+            try
+            {
+                source.CopyTo(new Span<char>(buffer));
+                Write(buffer, 0, source.Length);
+            }
+            finally
+            {
+                ArrayPool<char>.Shared.Return(buffer);
+            }
+        }
```

Comment:
Similarly, this is inefficient.  The implementation should look like that in `Write(string)`, and ideally if there's no measurable perf regression from doing so, they should share an implementation in a helper, with this doing a type check and delegating to the base if it's not a concrete StreamWriter.  The base already does this ArrayPool interaction.

Reply:
Yup got it! thanks! I'm reverting the whole stuff, reinstate the previous WriteLine overload and do perf measurement for the String overload using the two ways - original and delegation to ReadOnlySpan based overload...